x/concentrated-liquidity/incentives.go

```diff
@@ -415,8 +415,16 @@ func (k Keeper) GetUptimeGrowthOutsideRange(ctx sdk.Context, poolId uint64, lowe
 	return osmoutils.SubDecCoinArrays(globalUptimeValues, uptimeGrowthInside)
 }
 
-// initPositionUptime adds for all uptime accumulators `position` qualifies for.
-func (k Keeper) initPositionUptime(ctx sdk.Context, poolId uint64, owner sdk.AccAddress, lowerTick, upperTick int64, liquidityDelta sdk.Dec, joinTime time.Time, freezeDuration time.Duration, positionId uint64) error {
+// initOrUpdatePositionUptime either adds or updates records for all uptime accumulators `position` qualifies for
+func (k Keeper) initOrUpdatePositionUptime(ctx sdk.Context, poolId uint64, liquidity sdk.Dec, owner sdk.AccAddress, lowerTick, upperTick int64, liquidityDelta sdk.Dec, joinTime time.Time, positionId uint64) error {
+	// We update accumulators _prior_ to any position-related updates to ensure
+	// past rewards aren't distributed to new liquidity. We also update pool's
+	// LastLiquidityUpdate here.
+	err := k.updateUptimeAccumulatorsToNow(ctx, poolId)
+	if err != nil {
+		return err
+	}
+
```

Comment:
NOTE: I had to revert the `initPositionUptime` logic that was conflicting with this PR. The update position uptime logic is needed (as far as I could tell), despite getting rid of freezeDuration.

Reply:
In which cases do we now update position uptime? Is it if we withdraw early?

Reply:
Differing from our recent conversation in DMs, I actually think we need both the init and update branch of this function. Any time we withdraw a partial amount of liquidity (or even an entire amount for that matter), the positionCustomAccum should be updated with the liquidity delta.

The previous point I mentioned in DMs still stands, with regards to how checking the positionAge (in place of freezeDuration) does not make sense, it it should be the way it currently is in this PR IMO.